orchagent/flex_counter/dynamic_flex_counter_manager.h

```diff
@@ -0,0 +1,33 @@
+#ifndef SWSS_UTIL_DYNAMIC_FLEX_COUNTER_MANAGER_H
+#define SWSS_UTIL_DYNAMIC_FLEX_COUNTER_MANAGER_H
+
+#include "flex_counter_manager.h"
+
+// DynamicFlexCounterManager allows users to manage a group of flex counters
+// where the objects have highly variable sets of stats to track.
+class DynamicFlexCounterManager : public FlexCounterManager
```

Comment:
>DynamicFlexCounterManager [](start = 6, length = 25)

Name is confusing here. Is it dynamic counter or dynamic manager? For my reading, I guess it is `FlexCounterStatManager`

Reply:
Another option is to combine FlexCounterManager with DynamicFlexCounterManager into one class.

---
In reply to: [340903276](https://github.com/Azure/sonic-swss/pull/1093#discussion_r340903276) [](ancestors = 340903276)

Reply:
I'll go with your FlexCounterStatManager suggestion! The reason they're separate is related to this discussion: https://github.com/Azure/sonic-swss/pull/1093#issuecomment-548103642. Most of the existing daemons only set the stats once, so there's no need to maintain a mapping with all the objects and their stats.